src/main/java/frc/robot/Robot.java

```diff
@@ -17,32 +27,88 @@ public class Robot extends TimedRobot {
   //Class Initiation
   private final DriveTrain driveTrain = new DriveTrain();
   private final ControlInputs controlInputs = new ControlInputs();
+  private final SensorInputs sensorInputs = new SensorInputs();
+  private Components components = new Components();
+  private final ComponentsControl componentsControl = new ComponentsControl();
+
 
   //Variable Initiation
 
+  //Auto Variable Initiation
+  private String autoSelected;
+  private final String kAutoModeNull = "Do Nothing";
+  private final String kLeave = "Leave";
+  private final String kTurnTest = "Turn Test";
+  private ArrayList<AutoAction> autonomousSequence;
+  private SendableChooser<String> auto_chooser = new SendableChooser<String>();
+
   @Override
   public void robotInit() {
     //Drivetrain setup
     driveTrain.resetEncoders();
+
+    //Auto Chooser
+    auto_chooser.addOption(kAutoModeNull, kAutoModeNull);
+    auto_chooser.addOption(kLeave, kLeave);
+    auto_chooser.addOption(kTurnTest, kTurnTest);
+    auto_chooser.setDefaultOption(kAutoModeNull, kAutoModeNull);
+
+    SmartDashboard.putData("Auto Chooser", auto_chooser);
   }
 
   @Override
   public void robotPeriodic() {}
 
   @Override
-  public void autonomousInit() {}
+  public void autonomousInit() {
+    autonomousSequence = new ArrayList<AutoAction>();
+    autoSelected = auto_chooser.getSelected();
+    switch (autoSelected) {
+      case kLeave:
+        autonomousSequence.add(new AutoAction_MoveInline(5.0, 80.0, 2.0));
+        autonomousSequence.add(new AutoAction_DoNothing());
+        break;
+      case kTurnTest:
+        autonomousSequence.add(new AutoAction_Rotation(45.0, null));
+        autonomousSequence.add(new AutoAction_Rotation(-45.0, null));
+        autonomousSequence.add(new AutoAction_DoNothing());
+        break;
+      default:
+        autonomousSequence.add(new AutoAction_DoNothing());
+        break;
+    }
+    autonomousSequence.get(0).Init(driveTrain, components, sensorInputs);
+  }
 
   @Override
-  public void autonomousPeriodic() {}
+  public void autonomousPeriodic() {
+    if (autonomousSequence.size() > 0) {
+      SmartDashboard.putString("Current Auto Action",
+         autonomousSequence.get(0).getClass().toString());
```

Comment:
Rather than calling `getClass().toString()`, you should override the `toString()` method on each of the `AutoAction` classes, and call `autonomousSequence.get(0).toString()` directly. This is a bit more work, but you'll have a lot more control over how the logging works.